Add self-hosted Supabase tutorial series

khashashin opened the pull request:
I have read and understood the Contributor's Certificate of Origin available at the end of 
https://raw.githubusercontent.com/hetzneronline/community-content/master/tutorial-template.md
and I hereby certify that I meet the contribution criteria described in it.

Signed-off-by: Yusuf Khasbulatov khashashin@gmail.com

svenja11 commented:
Hello @khashashin, I was just testing your tutorial `deploy-supabase-coolify-hetzner` and I ran into an issue in Step 7. It says:

In Coolify:

1. Click Add New Project
2. Name it supabase-1
3. Click Add New Resource → Docker Compose → Supabase template

But this wasn't available for me. Did you write this tutorial for the latest Coolify version?

<img width="1604" height="599" alt="image" src="https://github.com/user-attachments/assets/c8f2ecee-9170-4bee-b435-eef2fbbf92eb" />

svenja11 commented:
Thank you for the quick update @khashashin 

I tested it again, this time both Supabase tutorials, and I noticed the following:

------

Deploy Self-Hosted Supabase on Hetzner Cloud with Coolify

* Step 7.3, can you please explain where exactly the domain should be added?
* Step 8 explains how to test the database connection. This did not work for me:
  ```bash
  psql "postgresql://postgres.<TENANT_ID>:<password>@supabase.<alphanumeric-string>.<your-domain.com>:5432/postgres"
  ```
  However, removing `<TENANT_ID>` worked for me:
  ```bash
  psql "postgresql://postgres:<password>@supabase.<alphanumeric-string>.<your-domain.com>:5432/postgres"
  ```

------

Harden PostgreSQL SSL for Self-Hosted Supabase

* Step 5.3, the file `/data/coolify/services/<SERVICE_UUID>/volumes/pooler/pooler.exs` doesn't exist for me. I checked and the service does exist:
  <img width="1309" height="174" alt="image" src="https://github.com/user-attachments/assets/79df9b35-14d7-4806-b0fe-e5dd56e8b610" />

khashashin commented:
Hi @svenja11, thanks for the feedback!

Step 7.3: I've clarified the instructions. Domain should be added in the **supabase-kong** service Settings → General tab → Domains field (or directly clicking on domain edit as shown on the screenshots).

Connection string: You're correct. The tenant ID prefix (postgres.<TENANT_ID>) only works after applying the SSL hardening tutorial. I've updated Steps 8.3 and 9 to use the correct format without the tenant ID, and added notes explaining this.

Regarding Step 5.3 (actually Step 5.4) the default file should be created by supavisor after successful start, then you should edit the default pooler.exs file by adding enforce_ssl attribute

svenja11 commented:
Thank you for always responding so quickly. I finally managed to test all of the tutorials. I have a few minor notes. Other than that, they look great and are ready from my side.

About my comment from above:

> Step 5.3, the file `/data/coolify/services/<SERVICE_UUID>/volumes/pooler/pooler.exs` doesn't exist for me.

I realised that it was a mistake from my side. I ran the commands on the Coolify server which is obviously wrong. I have now tested it on the Supabase server and everything worked as expected. Sorry about that 🙈 

------------

New notes:

* Deploy Self-Hosted Supabase on Hetzner Cloud with Coolify
  
  Step 8.3 >> I had to enable the public port in Coolify or else it couldn't connect: 
  * Select Supabase DB > Settings
  * Set "Public Port" to `5432`
  * Enable "Make it publicly available"
  
  However, this caused an issue in step 6.2 of the next tutorial "Harden PostgreSQL SSL for Self-Hosted Supabase".
  Before the deploy, I had to free that port again on the Supabase server:
  ```
  root@ubuntu-4gb-hel1-7:~# docker ps --format "table {{.ID}}\t{{.Names}}\t{{.Ports}}"
  CONTAINER ID   NAMES                                              PORTS
  8e08db8b09ee   sgwg0gso08s880o4sc8ckgwg-proxy                     80/tcp, 0.0.0.0:5432->5432/tcp, [::]:5432->5432/tcp
  
  root@ubuntu-4gb-hel1-7:~# docker stop sgwg0gso08s880o4sc8ckgwg-proxy
  root@ubuntu-4gb-hel1-7:~# docker rm sgwg0gso08s880o4sc8ckgwg-proxy
  ```
  Not sure what the best approach is here. Either not test the connection in the first tutorial or make sure the port is free at the beginning of the second tutorial?

<br>

* Centralized Security Monitoring with Prometheus and Grafana
  
  The final tutorial says to create a new server with Docker. It then says to connect to the server via SSH and how to create all the necessary files.
  
  In step 3 it says to restart Prometheus but since this is a newly created server, the output says "No such container: Prometheus"
  
  In step 7, `systemctl restart crowdsec` failed because `/etc/crowdsec/config.yaml` had two `prometheus:`:
  ```
  prometheus:
    enabled: true
    level: full
    listen_addr: 127.0.0.1
    listen_port: 6060
  
  prometheus:
    enabled: true
    level: full
    listen_addr: 0.0.0.0
    listen_port: 6060
  ```
  I removed the first Prometheus block, then it worked.
  
  In step 11 and 12, please specify on which server.
